### DISCUSSION TITLE
Modeling of ground source heat pump system

### DISCUSSION THREAD

**Discussion by ChaofanChen** [oemof/tespy#306] [Ideas]
Geothermal energy has gained more attention in recent years as a kind of renewable energy, especially for heating buildings. In order to practically and efficiently utilize the geothermal energy for building-heating purpose, ground source heat pump has to be coupled with the geothermal loop in the subsurface. 

Taking the closed-loop geothermal extraction system as an example, the outflow temperature of the closed-loop system changes dramatically during one heating season. It could be varied from 50 to 10 dC under a high heat load. This is a bad heat source for the heat pump, because the target fluid temperature for heating buildings needs to be designed within a fixed range, i.e., the design floor heating temperature should be kept at around 35 dC for buildings. 

So there is a question, is it needed to have a special heat pump technology to fully use the dynamic heat source from geothermal closed-loop system? Can this kind of ground source heat pump model be simulated by TESPy?

**Comment by fwitte** [comment]
@ChaofanChen, thank you for sharing your questions. I have not yet worked with such systems, I hope somebody else can help you here! How/why does the temperature vary that much? Does the subsurface temperature change that much? What about combinations of heating and cooling? That would also be a very good example application for TESPy, I think.

**Reply by ChaofanChen** [reply]
@fwitte thank you for the kindly reply. Yes, please see the field test data from Xi'an city in China in the following figure. It is a deep closed-loop borehole heat exchanger system in the subsurface ([see the reference](https://doi.org/10.1016/j.renene.2021.01.033)). 
![data](https://user-images.githubusercontent.com/23526219/154998126-7138f0ed-502e-4c33-9584-745e69b501f2.jpg)
Its outflow temperature (green dot line) can be decreased from around 43 dC to 20 dC over two months. During this period, the heat pump needs to be continuously operated. 
For such a deep closed-loop system, injecting heat is very hard because of the high geothermal temperature (initial geo-temperature at the bottom > 80 dC).

**Comment by jfreissmann** [comment]
Hey @ChaofanChen, first a disclaimer: I have never worked in building heat pumps, only modelling them.
In my opinion such a heat pump is feasible under certain conditions. The refrigerant has to have a working range containing the whole range between the lowest geothermal and the highest heating feed flow temperature (see for example the attached figure; taken from: Jesper et al. -- Large scale heat pumps: Uptake and performance modelling of market-available devices).

![grafik](https://user-images.githubusercontent.com/57762052/155091248-c7694470-5bb0-43ce-b27f-d4c532e988ec.png)

Besides that the compressor used in such a process would need to be able to dynamically alter the pressure ratio. Since the geothermal feed flow temperature does not seem to jump spontaneously, but rather change gradually, this seems plausible to me on a technical level. This would probably result in a broad range of COP though.

Concerning the modelling of such a heat pump using TESPy, my first idea would be to do a design calculation under "nominal" conditions and follow that up with offdesign calculations of the range of occuring geothermal feed flow temperatures. The result would be some kind of operating range depended of this temperature. But if you are looking to do transient simulations of heat pumps, TESPy is not able to offer that.

cc @maltefritz

**Reply by ChaofanChen** [reply]
Hi @jfreissmann, thank you so much for your instructive comment! 

The goal of the question is to achieve the coupled simulation between the geothermal closed-loop system and the heat pump system for heating buildings. In the most of the simulations of the ground source heat pump coupled closed-loop borehole heat exchanger system, the heat pump performance is always represented by a COP curve with a linear relationship with the BHE outflow (heat source) temperature, especially for shallow closed-loop system. However, this linear COP curve may not be suitable for the heat source from the deep closed-loop system presented above. 

> Concerning the modelling of such a heat pump using TESPy, my first idea would be to do a design calculation under "nominal" conditions and follow that up with offdesign calculations of the range of occuring geothermal feed flow temperatures. The result would be some kind of operating range depended of this temperature. 

This is very important logic. Thank you very much!

> But if you are looking to do transient simulations of heat pumps, TESPy is not able to offer that.

I think transient simulations can be achieved to some extent. This is because the simulation of the deep closed-loop geothermal system is transient, at every time step, the subsurface simulation can provide a heat source temperature for the coupled heat pump in TESPy with offdesign calculations.

**Reply by jfreissmann** [reply]
No worries, happy to help!

I think @fwitte has more experience regarding coupling of TESPy models and getting stable, converging solutions. But to me it seems plausible to calculate the heat pumps efficiency between each step of the quasi-transient geothermal simulation. Maybe you could pre-simulate every possible occuring feed flow temperature, so that you only had to look up the COP in each time step and reduce runtime.

> I think transient simulations can be achieved to some extent. This is because the simulation of the deep closed-loop geothermal system is transient, at every time step, the subsurface simulation can provide a heat source temperature for the coupled heat pump in TESPy with offdesign calculations.

I have not thought about doing that, but it seems very interesting. Are you doing a stationary simulation for each time step and use the results from the previous one as starting values? Would be awesome if you could share your work.

**Reply by ChaofanChen** [reply]
Thank you so much! 

Indeed, using the COP look-up table approach is an efficient way to achieved the coupled simulation. However, the important question is how the heat pump controls the subsurface closed-loop BHE system. 

For example, the different re-injection temperature and flow rate of the subsurface closed-loop will lead to a different outflow temperature after exchanging heat in subsurface. This is the heat source temperature that is acturally used by the heat pump. After using part of the heat by the heat pump under a certain amount of heat load from building site, there is a new re-injection temperature and flow rate to the closed-loop system that could be different from the previous ones. Therefore, it is the heat pump system that actually controls the subsurface loop according to its heating demand, the subsurface closed-loop responds accordingly. 

> Are you doing a stationary simulation for each time step and use the results from the previous one as starting values?

Yes, that is what I am thinking now for the heat pump part. We will work on this in the next step with @fwitte together. We would be happy to share the work after finishing it.

**Reply by fwitte** [reply]
Well, maybe @jfreissmann and @maltefritz would be a good fit as well :). Am quite busy right now, therefore some more help would be very helpful!

**Reply by ChaofanChen** [reply]
Yes, sure. That would be really helpful if they are interested, that's why the question is posted and discussed here. Thank you @fwitte to provide such a great discussion platform.

**Reply by fwitte** [reply]
More like thank GitHub :D, but I do think so too, should have set it up earlier :)

**Reply by maltefritz** [reply]
Hey @ChaofanChen, I am a colleague of @jfreissmann and we discussed your previous comments.

With what you describe in the second paragraph, it seems to us like it would be necessary to do a new simulation for each time step, so that you would get the correct temperatures for the subsurface loop. 

@jfreissmann and I are very interested in collaboration, especially regarding heat pumps and are looking forward to seeing your work.

**Reply by ChaofanChen** [reply]
Hi @maltefritz, thank you so much for your positive feedback! 

After finishing my PhD defense on 10.03.2022, I will focus on this work. Then we can arrange a short meeting together and prepare a shared file to better collaborate.
